src/puppetlabs/puppetdb/utils.clj

```diff
@@ -257,3 +257,10 @@
       {:href (to-href data)}
       (-> (sutils/parse-db-json data)
           (update :href to-href)))))
+
+(defn hsql?
+  "given the globals configuration bundle, determine if hsqldb is being used."
+  [globals]
```

Comment:
Basically, I could see functions like this being used when you only have access to `db` and not to `globals`.